### PR TITLE
NMEA Sentence upload

### PR DESCRIPTION
**Pull request by mfine**
This is the basics mechanics behind getting NMEA GGA sentences uploaded with libcurl - it does a trick to force READFUNC's to be called, and then uses pausing to rate limit the GGA uploads. I used a static global like @silverjam did because everything's in a single process space. Would need to get a cached NMEA GGA value in there to read up from.

/cc @anth-cole @silverjam

### CODE REVIEW

**Comment by anonymous-travis-swiftnav** [comment]
## v1.2.0-develop-2017122003-2-g61b5575
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2017122003-2-g61b5575/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.2.0-develop-2017122003-2-g61b5575/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2017122003-2-g61b5575

**Comment by anonymous-travis-swiftnav** [comment]
## v1.2.0-develop-2017122003-2-g61b5575
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2017122003-2-g61b5575/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.2.0-develop-2017122003-2-g61b5575/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2017122003-2-g61b5575

File: package/libnetwork/src/libnetwork.c [hunk 1]
```diff
@@ -390,7 +412,10 @@ void ntrip_download(const network_config_t *config)
 
   struct curl_slist *chunk = ntrip_init(curl);
 
+  curl_easy_setopt(curl, CURLOPT_PUT,              1L);
+  curl_easy_setopt(curl, CURLOPT_CUSTOMREQUEST,    "GET");
   curl_easy_setopt(curl, CURLOPT_URL,              config->url);
+  curl_easy_setopt(curl, CURLOPT_READFUNCTION,     network_download_read);
```
Comment (anth-cole) [inline package/libnetwork/src/libnetwork.c]: Should this be `network_upload_write`? Not sure if I'm understanding how the GGA put actually gets triggered

Comment (mfine) [reply package/libnetwork/src/libnetwork.c]: In libcurl, the direction goes the other way. The read functions means "give me something to read to upload". The write functions mean "I'm going to write something I've downloaded". So here this means give NMEA sentences to read that will get uploaded. I'd keep the `network_download_` prefix associated with the download function here (even though, confusingly, the download is doing some upload as well).

**Comment by silverjam** [comment]
This seems reasonable, but I'm assuming we need to implement some more hooks into the ntrip daemon so that it can receive the current position?

I think a difficult part of this implementation is going to be figuring out a reasonable IPC mechanism to allow the ntrip daemon to receive the data.  We can abuse signals again (maybe combined with shared memory) or we can try to integrate zmq and curl (with the more complicated curl interface).

Or, maybe the simplest solution is to have once of the Linux daemons that receives the POS_LLH message to write the current position to a file that the ntrip_daemon can read?

**Comment by mfine** [comment]
@silverjam yeah, that's the issue. I was assuming to do something minimal like what you already did for the base position in the rtcm decoder. I wouldn't invest time and energy into doing something more complicated - like the multi curl interface.

**Comment by anth-cole** [comment]
@silverjam @mfine I was thinking of putting in an sbp callback to listen for the POS_LLH message and then converting to the NMEA string, I couldn't see an easy way to listen for the NMEA GGA string being sent from the firmware, but I really don't have a good idea of how all this fits together, so open to suggestions

**Comment by mfine** [comment]
@anth-cole should be able to listen to the NMEA output by getting messages from the nmea routing:

https://github.com/swift-nav/piksi_buildroot/blob/v1.3.0-release/package/nmea_protocol/src/nmea_router.yml

listen to outgoing messages and cache the GGA messages (simple strcmp against the sentence). Should work the same as listening for POS_LLH messages but without having to do nmea encoding.

**Comment by silverjam** [comment]
Merged in #444, thanks @mfine

**Pull request closed without merging**